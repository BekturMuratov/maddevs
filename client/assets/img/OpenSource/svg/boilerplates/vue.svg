<svg xmlns="http://www.w3.org/2000/svg" width="336" height="125" fill="none" xmlns:v="https://vecta.io/nano"><g clip-path="url(#A)"><path d="M206.784.001l-38.797 67.203L129.191.001H-.008l167.996 290.997L335.984.001h-129.2z" fill="#41b883"/><path d="M206.785.001l-38.797 67.203L129.191.001h-62l100.798 174.595L268.786.001h-62.001z" fill="#34495e"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h336v291H0z"/></clipPath></defs></svg>