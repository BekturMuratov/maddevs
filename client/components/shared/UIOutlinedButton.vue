<template>
  <Component
    :is="isTag"
    :type="isLink ? 'button' : null"
    :href="linkTo"
    class="ui-outlined-button"
    :class="{
      'ui-outlined-button--grey': color === 'grey',
      'ui-outlined-button--black': color === 'black',
      'ui-outlined-button--red': color === 'red',
      'ui-outlined-button--blue': color === 'blue',
      'ui-outlined-button--full-width': fullWidth,
    }"
    @click="handleClick"
  >
    <slot />
  </Component>
</template>

<script>
export default {
  name: 'UIOutlinedButton',
  props: {
    color: {
      type: String,
      default: 'grey',
    },

    fullWidth: {
      type: Boolean,
      default: false,
    },

    isLink: {
      type: Boolean,
      default: false,
    },

    to: {
      type: String,
      default: '/',
    },
  },

  computed: {
    isTag() {
      if (this.isLink) return 'a'
      return 'button'
    },

    linkTo() {
      if (this.isLink) return this.to
      return null
    },
  },

  methods: {
    handleClick() {
      if (!this.disabled) this.$emit('click')
    },
  },
}
</script>

<style lang="scss" scoped>
.ui-outlined-button {
  @include font('Inter', 16px, 600);
  width: auto;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  background-color: transparent;
  box-shadow: none;
  border-radius: 4px;
  letter-spacing: -0.4px;
  padding: 16px 24px;
  line-height: 20px;
  border: 1px solid;
  transition: 0.4s;
  cursor: pointer;
  &--full-width {
    display: flex;
    width: 100%;
  }
  &--red {
    color: $text-color--red;
    border-color: $border-color--red;
    &:hover {
      background-color: $bgcolor--red;
      color: $text-color--white;
    }
  }
  &--blue {
    color: $text-color--blue;
    border-color: $border-color--blue;
    &:hover {
      background-color: $bgcolor--blue;
      color: $text-color--white;
    }
  }
  &--black {
    color: $text-color--black;
    border-color: $border-color--black;
    &:hover {
      background-color: $bgcolor--black;
      color: $text-color--white;
    }
  }
  &--grey {
    color: $text-color--quote-box;
    border-color: $border-color--grey-pale;
    &:hover {
      background-color: $border-color--grey-pale;
      color: $text-color--white;
    }
  }
}
</style>
