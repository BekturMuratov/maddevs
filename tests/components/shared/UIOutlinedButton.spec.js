import { render, screen } from '@testing-library/vue'
import UIOutlinedButton from '@/components/shared/UIOutlinedButton'

describe('UIOutlinedButton component', () => {
  it('should render correctly with slot', () => {
    const { container } = render(UIOutlinedButton, {
      slots: {
        default: 'Outlined button',
      },
    })

    expect(screen.getByText('Outlined button')).not.toBeNull()
    expect(container).toMatchSnapshot()
  })

  it('should render correctly without slot', () => {
    const { container } = render(UIOutlinedButton)

    expect(screen.queryByText('Outlined button')).toBeNull()
    expect(container).toMatchSnapshot()
  })
})
