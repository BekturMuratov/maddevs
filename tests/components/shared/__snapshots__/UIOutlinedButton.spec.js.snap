// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UIOutlinedButton component should render correctly with slot 1`] = `
<div>
  <button
    class="ui-outlined-button ui-outlined-button--grey"
    type="button"
  >
    Outlined button
  </button>
</div>
`;

exports[`UIOutlinedButton component should render correctly without slot 1`] = `
<div>
  <button
    class="ui-outlined-button ui-outlined-button--grey"
    type="button"
  />
</div>
`;
